src/torchcodec/encoders/_video_encoder.py

```diff
@@ -110,6 +121,9 @@ def to_tensor(
             pixel_format=pixel_format,
             crf=crf,
             preset=preset_value,
+            extra_options=[
+                x for k, v in (extra_options or {}).items() for x in (k, str(v))
+            ],
```

Comment:
I was going to suggest something like that instead of the previous `_flatten_codec_options` but refrained as I thought it might be too crazy lol.

I like it, I'd just suggest the following, which is a tiiiiiny bit safer, and makes it more obvious that everything in the list is a string.


```suggestion
            extra_options=[
                str(x) for k, v in (extra_options or {}).items() for x in (k, v)
            ],
```